src/app/shared/ProductItem/ProductItem.js

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable no-console */
+/* eslint-disable object-curly-newline */
+import React, { Component } from 'react';
+import { bool } from 'prop-types';
+
+import productType from '../../types';
+import HttpService from '../../../utils/http.service';
+import appConfig from '../../../config/appConfig';
+import { ViewFrontFull, ViewDetailsFull } from './viewFull';
+import { ViewCartSmall, ViewInfoSmall } from './viewSmall';
+import './ProductItem.scss';
+
+const CN = 'product-item';
+const { addToWishList, removeFromWishList } = appConfig.apiResources;
+
+class ProductItem extends Component {
+  static propTypes = {
+    extended: bool,
+
+    /**
+     * data - productType shape
+     */
+    data: productType
+  };
+
+  static defaultProps = {
+    extended: false,
+    data: {}
+  };
+
+  state = { showDetails: false };
+
+  showFront = () => this.setState({ showDetails: false });
+
+  showDetails = () => this.setState({ showDetails: true });
+
+  addToWishList = id => {
```

Comment:
move this method to ViewDetailsFull component

disable button `add/remove to wishlist` if http call is pending